src/interface/src/app/map/map-manager.ts

```diff
@@ -181,75 +181,77 @@ export class MapManager {
     }
   }
 
+  /**
+   * Adds a GeoJSON to the drawing layer.
+   * @param area: The geojson of the area to add to the drawing layer.
+   */
+  addGeoJsonToDrawing(area: GeoJSON.GeoJSON) {
+    L.geoJSON(area, {
+      style: (_) => ({
+        color: '#7b61ff',
+        fillColor: '#7b61ff',
+        fillOpacity: 0.2,
+      }),
+      pmIgnore: false,
+      onEachFeature: (_, layer) => {
+        layer.addTo(this.drawingLayer);
+        this.addClonedPolygons(layer);
+        layer.on('pm:edit', ({ layer }) => this.editHandler(layer));
+      },
+    });
+    this.polygonsCreated$.next(true);
+  }
+
+  /**
+   * Given the original polygon, adds the cloned polygons to the cloned drawing
+   * layer on all maps.
+   */
+  private addClonedPolygons(layer: L.Layer) {
+    this.maps.forEach((currMap) => {
+      const originalId = L.Util.stamp(layer);
+
+      // Hacky way to clone, but it removes the reference to the origin layer
+      const clonedLayer = L.geoJson((layer as L.Polygon).toGeoJSON()).setStyle({
+        color: '#ffde9e',
+        fillColor: '#ffde9e',
+      });
+      currMap.clonedDrawingRef?.addLayer(clonedLayer);
+      currMap.drawnPolygonLookup![originalId] = clonedLayer;
+    });
+  }
+
+  /**
+   * Given the original polygon, removes the cloned polygons from the cloned drawing
+   * layer on all maps. Optionally deletes the original polygon key.
+   */
+  private removeClonedPolygons(layer: L.Layer, toDelete: boolean) {
```

Comment:
supernit: deleteOriginal instead of toDelete?

Reply:
yeah that's much clearer. done!